docs/content/standards/coin.mdx

```diff
@@ -67,72 +75,47 @@ The fields of the metadata objects include the following:
-| `deny_cap_object`      | The ID of the token's `DenyCap` object, which is necessary to maintain the deny list entries that controls who can and cannot use the token. |
+| `deny_cap_object`      | The ID of the token's `DenyCapV2` object, which is necessary to maintain the deny list entries that controls who can and cannot use the token. |
 
 ## Minting and burning coins
 
 The `coin` module provides the logic for creating and destroying coins on the Sui network (as long as you own the associated `TreasuryCap`). These functions are the same for all coins and each requires the `TreasuryCap` as an input.
 
 ### Mint
 
-Use the `coin::mint` function to create new tokens. The signature for the function is:
+Use the `coin::mint` function to create new tokens.
 
-```move
-public fun mint<T>(
-  cap: &mut coin::TreasuryCap<T>,
-  value: u64,
-  ctx: &mut tx_context::TxContext
-): coin::Coin<T>
-```
+{@inject: crates/sui-framework/packages/sui-framework/sources/coin.move#fun=mint noComments}
 
 The signature shows that a `Coin<T>` results from calling the function with a `TreasuryCap`, value for the coin created, and the transaction context. The function updates the total supply in `TreasuryCap` automatically. Upon display, the coin `value` respects the `decimals` value in the metadata. So, if you supply 1000000 as the coin `value` that has a `decimal` value of `6`, the coin's value displays as `1.000000`.
 
 ### Burn
 
-Use the `coin::burn` function to destroy current tokens. The signature for the function is:
+Use the `coin::burn` function to destroy current tokens.
 
-```move
-public entry fun burn<T>(
-  cap: &mut coin::TreasuryCap<T>,
-  c: coin::Coin<T>
-): u64
-```
+{@inject: crates/sui-framework/packages/sui-framework/sources/coin.move#fun=burn noComments}
 
 The signature shows that only the `TreasuryCap` and coin object you want to burn are necessary inputs, returning the amount by which the supply was decreased (value of the coin). The function does not allow you to burn more coins than are available in the supply.
 
 ## Adding and removing addresses to and from the deny list
 
-The deny list is only applicable to regulated coins. As mentioned previously, when you create a regulated coin you receive a `DenyCap` that authorizes the bearer to add and remove addresses from the system-created `DenyList` object. Any address on the list for your coin is unable to use the coin as an input to transactions.
+The deny list is only applicable to regulated coins. As mentioned previously, when you create a regulated coin you receive a `DenyCapV2` that authorizes the bearer to add and remove addresses from the system-created `DenyList` object. Any address on the list for your coin is unable to use the coin as an input to transactions starting immediately upon being added. At the following epoch after addresses get added to the list, those addresses cannot receive the coin type.
```

Comment:
```suggestion
The deny list is only applicable to regulated coins. As mentioned previously, when you create a regulated coin you receive a `DenyCapV2` that authorizes the bearer to add and remove addresses from the system-created `DenyList` object. Any address on the list for your coin is unable to use the coin as an input to transactions starting immediately upon being added. At the following epoch after addresses get added to the list, those addresses additionally cannot receive the coin type.
```
Thoughts on making it clearer that this is additional behavior? i.e. that it is not that it somehow switches from sending to receiving.

Reply:
Good call. I also like the general "sending" and "receiving" language, so I additionally added it.